plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/GcpLocalRunTab.java

```diff
@@ -98,15 +115,18 @@
   private boolean activated;
 
   public GcpLocalRunTab(EnvironmentTab environmentTab) {
-    this(environmentTab, PlatformUI.getWorkbench().getService(IGoogleLoginService.class),
+    this(environmentTab,
+        PlatformUI.getWorkbench().getService(IGoogleLoginService.class),
+        PlatformUI.getWorkbench().getService(IGoogleApiFactory.class),
         new ProjectRepository(PlatformUI.getWorkbench().getService(IGoogleApiFactory.class)));
```

Comment:
I wonder if we should push this service-fetching into the tab group.

Reply:
Thought about that, and my thinking has been that, if a class surely knows how to get something, why pass the burden of acquiring it to callers, unless there is a merit (there doesn't seem to me)? I may be wrong, but this is a UI class, so I thought `PlatformUI.getWorkbench()` must always succeed. (Otherwise, how one could possibly show a UI tab?)

Reply:
The downside is that it makes and bakes in an assumption about where these instances are from.  So it's kinda-injected, but not quite.  Anyways, this is done elsewhere in the code and it's something that can be tackled later.